### PR TITLE
Send cum{sum, prod} to CPU for long type

### PR DESCRIPTION
**Pull request by asuhan**

### CODE REVIEW

File: torch_xla/csrc/aten_xla_type.cpp [hunk 1]
```diff
@@ -945,24 +945,42 @@ at::Tensor AtenXlaType::cross(const at::Tensor& self, const at::Tensor& other,
 
 at::Tensor AtenXlaType::cumprod(const at::Tensor& self, int64_t dim,
                                 at::ScalarType dtype) const {
-  return bridge::AtenFromXlaTensor(
-      XLATensor::cumprod(bridge::GetXlaTensor(self), dim, dtype));
+  XLATensor self_tensor = bridge::GetXlaTensor(self);
```
Comment (dlibenzi) [inline torch_xla/csrc/aten_xla_type.cpp]: In tensor_util.h:
```
inline bool HasNativeSupport(xla::PrimitiveType type, const Device& device) {
  return GetDevicePrimitiveType(type, &device) == type;
}
```

And use that.

**Comment by dlibenzi** [comment]
I am also wondering, %reduce seems to support S64.
Are there cases where we are reducing the full window, and hence we can use %reduce instead?

File: torch_xla/csrc/aten_xla_type.cpp (hunk 1 above)
Comment (asuhan) [reply torch_xla/csrc/aten_xla_type.cpp]: We don't have an `xla::PrimitiveType` here, we have an `at::ScalarType`. How would this work?

**Comment by asuhan** [comment]
I don't see a way, a reduce only returns the complete result. We could at most do it with many reduces, which I don't think would be a good idea.

File: torch_xla/csrc/aten_xla_type.cpp (hunk 1 above)
Comment (dlibenzi) [reply torch_xla/csrc/aten_xla_type.cpp]:
```
inline bool HasNativeSupport(at::ScalarType type, const Device& device) {
  return TensorTypeFromXlaType(MakeXlaPrimitiveType(type, &device)) == type;
}
```

Comment (asuhan) [reply torch_xla/csrc/aten_xla_type.cpp]: Obviously! Ok.

Comment (asuhan) [reply torch_xla/csrc/aten_xla_type.cpp]: Done.

**Pull request merged**